src/neuroconv/datainterfaces/ecephys/basesortingextractorinterface.py

```diff
@@ -226,6 +226,69 @@ def subset_sorting(self):
         stub_sorting_extractor = self.sorting_extractor.frame_slice(start_frame=0, end_frame=end_frame)
         return stub_sorting_extractor
 
+    def add_channel_metadata_to_nwb(self, nwbfile: NWBFile, metadata: Optional[DeepDict] = None):
+        """
+        Add channel metadata to an NWBFile object using information extracted from a SortingExtractor and
+        optional metadata.
+
+        This function attempts to add devices, electrode groups, and electrodes to the NWBFile. If a recording is
+        associated with the SortingExtractor, it is used for metadata addition. Otherwise, it attempts to create a dummy
+        NumpyRecording based on the provided metadata. If neither is available, the function warns the user and skips the
+        metadata addition.
+
+        Parameters
+        ----------
+        nwbfile : NWBFile
+            The NWBFile object to which the metadata is added.
+        metadata : Optional[DeepDict]
+            Optional metadata to use for the addition of electrode-related data. If it's provided, it should contain an
+            "Ecephys" field with a nested "Electrodes" field.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        Warning
+            If there's no recording in the sorting extractor and no electrodes metadata in the provided metadata, a warning
+            is raised and the function returns None.
+
+        Notes
+        -----
+        This function adds metadata to the `nwbfile` in-place, meaning the `nwbfile` object is modified directly.
+        """
+        from ...tools.spikeinterface import (
+            add_devices,
+            add_electrode_groups,
+            add_electrodes,
+        )
+
+        if hasattr(self, "generate_recording_with_channel_metadata"):
+            recording = self.generate_recording_with_channel_metadata()
+        else:
+            electrodes_metadata = metadata.get("Ecephys", dict()).get("Electrodes", list())
+            if electrodes_metadata:
+                n_channels = max([len(x["data"]) for x in metadata["Ecephys"]["Electrodes"]])
+                from spikeinterface import NumpyRecording
+
+                recording = NumpyRecording(
+                    traces_list=[np.empty(shape=(1, n_channels))],
+                    sampling_frequency=self.sorting_extractor.get_sampling_frequency(),
+                )
+            else:
+                import warnings
+
+                warnings.warn(
+                    "No recording found in sorting extractor or electrodes_metadata in the metadata metadata"
+                    "Skipping addition of recording metadata to the NWBFile."
+                )
+                return None
```

Comment:
```suggestion
                return
```

Any reason to include `None`? Kind of like annotating `-> None`, isn't it just the built-in default?

Reply:
I feel that is more explicit and easier to infer that is an early termination of the function but I don't feel very strong about this. Do you prefer the empty return?

Anyway, as said above, this else block probably should go away anyway...

Reply:
I feel that is more explicit and easier to infer that is an early termination of the function but I don't feel very strong about this. Do you prefer the empty return?

Anyway, as said above, this else block probably should go away anyway...

Reply:
Pythonic explicit > implicit, I'm just wondering where the line is drawn 😆

Technically if we want to be super explicit we should fill every function call or initialization with all of their keywords, even the ones we  set to default; I see  `return None` as being at that same category, since the default of `return` is `return None` and the default of `def()` is `def() -> None`

Not a huge deal no matter

Reply:
Yeah, it goes either way to me, same with annotation -> None for `__init__`. 

The else block was removed anyway.